src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/SupplierAssessmentService.kt

```diff
@@ -0,0 +1,87 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Appointment
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthUser
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Referral
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.SupplierAssessment
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.AppointmentRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.AuthUserRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ReferralRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.SupplierAssessmentRepository
+import java.time.OffsetDateTime
+import java.util.UUID
+import javax.persistence.EntityNotFoundException
+import javax.transaction.Transactional
+
+@Service
+@Transactional
+class SupplierAssessmentService(
+  val supplierAssessmentRepository: SupplierAssessmentRepository,
+  val referralRepository: ReferralRepository,
+  val appointmentRepository: AppointmentRepository,
+  val authUserRepository: AuthUserRepository,
+) {
+  fun createSupplierAssessment(
+    referral: Referral,
+  ): Referral {
+    referral.supplierAssessment = supplierAssessmentRepository.save(
+      SupplierAssessment(
+        id = UUID.randomUUID(),
+        referral = referral,
+      )
+    )
+    return referralRepository.save(referral)
+  }
+
+  fun scheduleOrUpdateSupplierAssessmentAppointment(
+    supplierAssessmentId: UUID,
+    durationInMinutes: Int,
+    appointmentTime: OffsetDateTime,
+    createdByUser: AuthUser
+  ): Appointment {
+    val supplierAssessment = getSupplierAssessmentById(supplierAssessmentId)
+
+    if (supplierAssessment.appointments.size == 0) {
```

Comment:
Why is this needed? Isn't the appointment always in existence once referral is sent?

Reply:
The supplier assessment gets created. The appointment is not actually created until the schedule endpoint is called for the first time.